packages/dev/docs/pages/blog/building-a-combobox.mdx

```diff
@@ -0,0 +1,94 @@
+As of INSERT DATE HERE, React Spectrum ComboBox has been released as an alpha. For those who may be unfamiliar with a combobox, it is an input field that has a popover menu associated with it. The popover menu contains
+a list of values within that a user may select as the value for the combobox and is often filtered to display possible matches to the user's current
+input text. Finally, there is a button accompanying the input field used for toggling the open state of the popover menu.
+
+While the above may seem rather straight forward, a combobox contains a significant amount of complexity. Focus needs to remain within the input field
+regardless of the popup menu's open state so that a user may continue to filter down the list, but virtual focus needs to be displayed so that items
+can be highlighted and selected via keyboard commands as well. A combobox's input value will often diverge and converged from its currently selected value based on
+the user's interaction and the combobox's configuration. Cross browser and device support adds additional layers of complexity with each of their behavioral quirks and
+interaction models.
+
+In this blog post, we'll be covering the challenges we encountered with regards to ComboBox's mobile experience and accessibility.
+
+## Mobile experience
+
+Like many of our other components, ComboBox displays its popover menu items in a tray when rendered on a mobile device or a smaller screen. This allows for a better mobile experience
+since the tray can display more items on the screen and grants users a larger hit area to scroll through. To compensate for the fact that the tray covers the majority of the
+screen including the ComboBox, we included an input field within the tray so that end users could still type to filter the available items. So far so good. However,
+when we tested the ComboBox's tray input on iOS Safari, the tray's bottom half was covered by the onscreen keyboard and we could not scroll the hidden items into view.
+
+It turns out that iOS Safari does not shrink the browser window to accommodate the onscreen keyboard unlike other mobile browsers. Instead, Safari pushes the window upwards and
+partially off screen. At the time, we had been relying on `window.innerHeight` to inform us on how tall the tray should be but now we had to find a different way of tracking the available window
+space for the tray.
+
+Luckily for us, iOS 13 added support for the VisualViewport API. By querying `window.VisualViewport.height` we could get a reliable measurement of how much vertical space was available on screen. Furthermore, we could track
+when the onscreen keyboard was opened or dismissed by listening to the VisualViewport's `resize` event. Leveraging these two allowed us to create a tray that properly adjusts to the presence of iOS onscreen keyboard.
+If you'd like to track the visual viewport size in your own app, you can use the `useViewportSize` hook available in the `@react-aria/utils` package.
+
```

Comment:
I suppose we should also cover the challenges in getting mobile Safari to not scroll the window when focusing the input? The comment in usePreventScroll covers it pretty well.

Reply:
I can mention preventing page scrolling so that the tray itself can't be scrolled up/down when the keyboard is expanded, but I can't repro the following comment in the combobox after commenting out the fix:

```
3. When tapping on an input, the page always scrolls so that the input is centered in the visual viewport.
    This may cause even fixed position elements to scroll off the screen.
``` 

I think the above was perhaps specific to if the input is near the bottom of the screen prior to the onscreen keyboard being opened (can only repro in the dialog with long content story)